### DISCUSSION TITLE
Any advice would be appreciated. - DATA_MSG, DATA_VIRTUAL

### DISCUSSION THREAD

**Discussion by jongwookim1020** [jovibor/HexCtrl#31] [Q&A]
Thank you in advance.   ^___^

-. My work environment
   Windows 10, Visual Studio 2017 (v141)

-. about "HexCtrl - DATA_MSG, DATA_VIRTUAL" 
   Can you give me some examples of usage in simple code?
   I tried to do it based on your explanation, I am often not successful.

**Comment by jovibor** [comment]
You have to be more specific. What exactly do you have issues with?

You can take a look at the [Pepper](https://github.com/jovibor/Pepper) app, it features exactly `DATA_MSG` approach.
The message handling code can be found [here](https://github.com/jovibor/Pepper/blob/953670692c514678d634f20a80c72b6d050955d8/Pepper/FileLoader.cpp#L280)

**Reply by jongwookim1020** [reply]
![test](https://user-images.githubusercontent.com/76579296/103162234-75177100-4831-11eb-8958-bf10e5c3a3ad.jpg)

-. function
    if i click HexCtrl, 
    1) editbox1 = Caret .  it will be clear using "m_myHex->GetCaretPos()" + HEXCTRL_MSG_SETSELECTION.
    2) editbox2 = Data.    what can i do for this function? DATA_VIRTUAL? + HEXCTRL_MSG_SETSELECTION.
                                       I read about Virtual Handler, but I don't know how to use it.

**Reply by jovibor** [reply]
You want to get a data (byte) from a given offset, am i right?
If so, well, data is always external to **HexCtrl**, you set it in `SetData` method. 
So, you already have your data pointer somewhere, just use `ptrYourData + m_myHex->GetCaretPos();`

Alternatively you can use [`m_myHex->GetData( { m_myhex->GetCaretPos(), 1 } );`](https://github.com/jovibor/HexCtrl#getdata) interface method.
